### DISCUSSION TITLE
There is no secondary storage VM for downloading template to image store

### DISCUSSION THREAD

**Discussion by dylan62370** [apache/cloudstack#9535] [Q&A]
Hello ! :)

Unfortunately, I can't transfer ISO/Templates to my cloudstack.

When I transfer locally or from register, I get this error message:
“Request failed (530)
There is no secondary storage VM for downloading template to image store NFS01”.
<img width="394" alt="image" src="https://github.com/user-attachments/assets/e907e65c-1ac5-4f8b-b3e5-9e31d7f75bba">

My NFS01 seems to be working properly.
On my cloud VM (where cloudstack-management is stored)

I made this command:
sudo mount -t nfs 10.1.0.200:/export/secondary /mnt/secondary

To test whether it mounts properly on my cloud VM and it mounts without a problem, I can easily create folders and files that fit into /export/secondary on the NFS.

PS: the IP 10.1.0.200 is the IP of my NFS server (NFS01)

Translated with DeepL.com (free version)

Here is my NFS assembly:
<img width="1143" alt="image" src="https://github.com/user-attachments/assets/25a43d9d-58af-4d09-98fa-0a7e971a7388">

And here is my “aud-fr” zone
<img width="787" alt="image" src="https://github.com/user-attachments/assets/402a9454-e545-43af-9c73-c53a05245889">

CloudStack 4.19.1.1
CentOS 9 (vm cloud and kvm server)

I would be very grateful for any help you could give me, as I have no idea what to do...

**Comment by DaanHoogland** [comment]
@dylan62370 , can you check "infrastructure/system vms"? you should see something like 
![image](https://github.com/user-attachments/assets/dc6aea35-226f-4b1f-887e-7cea229d6979)
if the secondarystoragevm is not there or its agent state is not up, you need to look at that first. 
if the ssvm is there you can look into it why it doesn't function.
if it is not there you'll need to look at the management server log as to why it won't deploy.

**Comment by dylan62370** [comment]
Hello @DaanHoogland !

The VM is UP but I've just seen that the agent is DOWN...
I don't understand why...

<img width="1166" alt="image" src="https://github.com/user-attachments/assets/44c9c564-0868-400f-a557-ebfb764c06cd">

**Comment by rajujith** [comment]
@dylan62370  Since the VM is running but the agent is down this indicates an issue for the java agent in the system VMs to connect to the management server port TCP port 8250. Can you check the following:
1) Check the /var/log/cloud.log on the system VM.
2) Check connectivity to the management server from the system VM.

**Reply by dylan62370** [reply]
Hello, @rajujith 

I found the problem, my network was not good.
I redid everything and the agent is now UP.

However, when I want to download a template or an image I encounter this error for example for Debian 12:

Unable to resolve cdimage.debian.org

When I ping cdimage.debian.org from the cloudstack management VM I ping google without problem...

![image](https://github.com/user-attachments/assets/4875f565-8565-4427-aa87-e7189fe77d9d)

**Reply by DaanHoogland** [reply]
@dylan62370 , try logging into the ssvm and ping from there. Also check the dns settings (resolv.conf) on the ssvm. 
( run /usr/local/cloudstack/systemvm/ssvm-check.sh and examine the output )

**Reply by dylan62370** [reply]
@DaanHoogland 

The ping to google is KO the ping to my gateway 10.0.0.1 is KO too on the ssvm.

```
root@s-2-VM:~# ping 1.1.1.1
PING 1.1.1.1 (1.1.1.1): 56 data bytes
92 bytes from s-2-VM (10.0.0.181): Destination Host Unreachable
92 bytes from s-2-VM (10.0.0.181): Destination Host Unreachable
92 bytes from s-2-VM (10.0.0.181): Destination Host Unreachable
92 bytes from s-2-VM (10.0.0.181): Destination Host Unreachable
```

When I go to the /etc/network/interfaces interface of the ssvm I notice that I don't have any gateway, is this normal?

SSVM Network
```
auto lo eth0 eth1 eth2
iface lo inet loopback

iface  eth0 inet static
  address 169.254.71.94
  netmask 255.255.0.0
iface  eth1 inet static
  address 10.0.0.62
  netmask 255.255.255.0
iface  eth2 inet static
  address 10.0.0.181
  netmask 255.255.255.0
```
```
  ip route
  default via 10.0.0.1 dev eth2 
10.0.0.0/24 dev eth1 proto kernel scope link src 10.0.0.62 
10.0.0.0/24 dev eth2 proto kernel scope link src 10.0.0.181 
10.1.0.9 via 10.0.0.1 dev eth1 
169.254.0.0/16 dev eth0 proto kernel scope link src 169.254.71.94 
```
  
  
<img width="1124" alt="image" src="https://github.com/user-attachments/assets/e27307a4-9952-4209-9dc5-a85281f8c12b">

Network cloudstack management + KVM:

```
[dylan@cloudstack-test ~]$ ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: ens33: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq master cloudbr0 state UP group default qlen 1000
    link/ether 00:50:56:84:33:70 brd ff:ff:ff:ff:ff:ff
    altname enp2s1
3: cloudbr0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 00:50:56:84:33:70 brd ff:ff:ff:ff:ff:ff
    inet 10.0.0.22/24 brd 10.0.0.255 scope global noprefixroute cloudbr0
       valid_lft forever preferred_lft forever
    inet6 fe80::9897:46f5:48ea:f99c/64 scope link noprefixroute 
       valid_lft forever preferred_lft forever
5: cloud0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether fe:00:a9:fe:47:5e brd ff:ff:ff:ff:ff:ff
    inet 169.254.0.1/16 scope global cloud0
       valid_lft forever preferred_lft forever
    inet6 fe80::dcd1:e6ff:fed4:fabb/64 scope link 
       valid_lft forever preferred_lft forever
6: vnet0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master cloud0 state UNKNOWN group default qlen 1000
    link/ether fe:00:a9:fe:47:5e brd ff:ff:ff:ff:ff:ff
    inet6 fe80::fc00:a9ff:fefe:475e/64 scope link 
       valid_lft forever preferred_lft forever
7: vnet1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master cloudbr0 state UNKNOWN group default qlen 1000
    link/ether fe:00:2c:00:00:03 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::fc00:2cff:fe00:3/64 scope link 
       valid_lft forever preferred_lft forever
8: vnet2: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master cloudbr0 state UNKNOWN group default qlen 1000
    link/ether fe:00:26:00:00:0c brd ff:ff:ff:ff:ff:ff
    inet6 fe80::fc00:26ff:fe00:c/64 scope link 
       valid_lft forever preferred_lft forever
9: vnet3: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master cloud0 state UNKNOWN group default qlen 1000
    link/ether fe:00:a9:fe:4e:8b brd ff:ff:ff:ff:ff:ff
    inet6 fe80::fc00:a9ff:fefe:4e8b/64 scope link 
       valid_lft forever preferred_lft forever
10: vnet4: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master cloudbr0 state UNKNOWN group default qlen 1000
    link/ether fe:00:fd:00:00:0a brd ff:ff:ff:ff:ff:ff
    inet6 fe80::fc00:fdff:fe00:a/64 scope link 
       valid_lft forever preferred_lft forever
11: vnet5: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master cloudbr0 state UNKNOWN group default qlen 1000
    link/ether fe:00:b2:00:00:0b brd ff:ff:ff:ff:ff:ff
    inet6 fe80::fc00:b2ff:fe00:b/64 scope link 
       valid_lft forever preferred_lft forever
```
```
root@s-2-VM:~# /usr/local/cloud/systemvm/ssvm-check.sh
================================================
First DNS server is  10.0.0.15
PING 10.0.0.15 (10.0.0.15): 56 data bytes
92 bytes from 10.0.0.62: Destination Host Unreachable
92 bytes from 10.0.0.62: Destination Host Unreachable
--- 10.0.0.15 ping statistics ---
2 packets transmitted, 0 packets received, 100% packet loss
WARNING: cannot ping DNS server
route follows
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
0.0.0.0         10.0.0.1        0.0.0.0         UG    0      0        0 eth2
10.0.0.0        0.0.0.0         255.255.255.0   U     0      0        0 eth1
10.0.0.0        0.0.0.0         255.255.255.0   U     0      0        0 eth2
10.1.0.9        10.0.0.1        255.255.255.255 UGH   0      0        0 eth1
169.254.0.0     0.0.0.0         255.255.0.0     U     0      0        0 eth0
================================================
ERROR: DNS not resolving cloudstack.apache.org
resolv.conf follows
nameserver 10.0.0.15
nameserver 10.1.0.9
nameserver 1.1.1.1
nameserver 8.8.8.8
root@s-2-VM:~# 
```

For the test I've installed KVM and cloudstack-management on the same server for the time being.
I followed this tutorial
https://www.youtube.com/watch?v=U7RLzXXemCY

**Reply by DaanHoogland** [reply]
@dylan62370 , all looks good except for your dns. Does it delegate any unknown domains from 10.0.0.15 and 10.1.0.9? or is it caching ok?

**Reply by dylan62370** [reply]
@DaanHoogland 

All my servers use my internal DNS.
I can use them to reach Google and other sites without worry.

However, from the SSVM VM I can't even ping them. 
Cloudstack-management uses these DNS and it works fine, I can ping 10.0.0.15

And from my PC I ping the cloudstack IP which is 10.0.0.22 but not the SSVM VM IP 10.0.0.62 without any pr

**Reply by DaanHoogland** [reply]
so can you ping 8.8.8.8 (and 1.1.1.1) from the ssvm?

**Reply by dylan62370** [reply]
@DaanHoogland 
No it doesn't work, I rebuilt the VM to see if it was coming from the server but no.

The ping still doesn't go through from the SSVM VM.

```
root@s-2-VM:~# ping 1.1.1.1
PING 1.1.1.1 (1.1.1.1): 56 data bytes
92 bytes from s-2-VM (10.0.0.181): Destination Host Unreachable
92 bytes from s-2-VM (10.0.0.181): Destination Host Unreachable
92 bytes from s-2-VM (10.0.0.181): Destination Host Unreachable
92 bytes from s-2-VM (10.0.0.181): Destination Host Unreachable
^C--- 1.1.1.1 ping statistics ---
5 packets transmitted, 0 packets received, 100% packet loss
root@s-2-VM:~# ping 8.8.8.8
PING 8.8.8.8 (8.8.8.8): 56 data bytes
92 bytes from s-2-VM (10.0.0.181): Destination Host Unreachable
92 bytes from s-2-VM (10.0.0.181): Destination Host Unreachable
```

**Reply by DaanHoogland** [reply]
could you check the new VM also and look at the routing table and also look at the traceroute output?
In you routing table I see the route for 10.0.0.0 twice and on different intefaces. Do two of your physical networks have the same range? You might want to define them as one physical network with two traffic types, if so.

**Reply by dylan62370** [reply]
> @dylan62370 Since the VM is running but the agent is down this indicates an issue for the java agent in the system VMs to connect to the management server port TCP port 8250. Can you check the following:
> 
>     1. Check the /var/log/cloud.log on the system VM.
> 
>     2. Check connectivity to the management server from the system VM.

@rajujith  this is log
![image](https://github.com/user-attachments/assets/dd106e09-c750-4508-8d7e-e6406832adb6)


The ping is OK:  
10.0.0.22 -> 10.0.0.181 (Public IP address) - > ssvpm
10.0.0.22 -> 169.254.26.185 (Link-local/Control IP address) -> ssvpm
10.0.0.22 -> 10.0.0.66 (Private IP address) -> ssvm

10.0.0.22 -> 10.0.0.180 (Public IP address) -> consoleproxy
10.0.0.22 -> 169.254.28.60 (Link-local/Control IP address) -> consoleproxy
10.0.0.22 -> 10.0.0.64 (Private IP address) ->consoleproxy

**Comment by dylan62370** [comment]
@DaanHoogland 

I currently have 1 active network card on the VM.
This VM contains KVM, NFS and cloudstack-management.

(this is a test before moving on to a real network)

So I made a cloudb0 bridge which is bridged on the ens33 network card.

And I see that I have two routes to eth1 and eth2.
Should I delete one of the routes?

```
root@s-2-VM:~# ip route
default via 10.0.0.1 dev eth2 
10.0.0.0/24 dev eth1 proto kernel scope link src 10.0.0.66 
10.0.0.0/24 dev eth2 proto kernel scope link src 10.0.0.181 
10.1.0.9 via 10.0.0.1 dev eth1 
169.254.0.0/16 dev eth0 proto kernel scope link src 169.254.26.185 
```

Or should I add a route from my gateway (10.0.0.1) to my DNS, for example (10.0.0.15)?

**Reply by DaanHoogland** [reply]
@dylan62370 , I am not sure but if you have only one IF on the VM it shouldn't see two (and create a second route for that)
Though it looks like the routes from that last table you send are not conflicting, try deleting the line
```10.0.0.0/24 dev eth2 proto kernel scope link src 10.0.0.18```

Can you try `traceroute 1.1.1.1` and/or `traceroute 8.8.8.8`?
and post the outut of `ip a `?
Did you run /usr/local/cloud/systemvm/ssvm-check.sh yet?

**Comment by dylan62370** [comment]
@DaanHoogland 

I removed the route and here's the result of the commands:
The commands were made on the SSVM VM

**ip a** 
```
root@s-2-VM:~# ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether 0e:00:a9:fe:1a:b9 brd ff:ff:ff:ff:ff:ff
    altname enp0s3
    altname ens3
    inet 169.254.26.185/16 brd 169.254.255.255 scope global eth0
       valid_lft forever preferred_lft forever
3: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether 1e:00:11:00:00:07 brd ff:ff:ff:ff:ff:ff
    altname enp0s4
    altname ens4
    inet 10.0.0.66/24 brd 10.0.0.255 scope global eth1
       valid_lft forever preferred_lft forever
4: eth2: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether 1e:00:26:00:00:0c brd ff:ff:ff:ff:ff:ff
    altname enp0s5
    altname ens5
    inet 10.0.0.181/24 brd 10.0.0.255 scope global eth2
       valid_lft forever preferred_lft forever
```

**Traceroute:**
```
root@s-2-VM:~# traceroute 1.1.1.1
traceroute to 1.1.1.1 (1.1.1.1), 30 hops max, 60 byte packets
 1  s-2-VM (10.0.0.181)  3053.751 ms !H  3053.676 ms !H  3053.657 ms !H
root@s-2-VM:~# traceroute 8.8.8.8
traceroute to 8.8.8.8 (8.8.8.8), 30 hops max, 60 byte packets
 1  * * *
 2  * * *
 3  * * *
 4  * * *
 5  * * *
 6  * * *
 7  * * *
 8  * * *
 9  * * *
10  * * *
11  * * s-2-VM (10.0.0.181)  2207.822 ms !H
```

**ssvm-check:**
```
root@s-2-VM:~# /usr/local/cloud/systemvm/ssvm-check.sh
================================================
First DNS server is  10.0.0.15
PING 10.0.0.15 (10.0.0.15): 56 data bytes
92 bytes from 10.0.0.66: Destination Host Unreachable
92 bytes from 10.0.0.66: Destination Host Unreachable
--- 10.0.0.15 ping statistics ---
2 packets transmitted, 0 packets received, 100% packet loss
WARNING: cannot ping DNS server
route follows
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
0.0.0.0         10.0.0.1        0.0.0.0         UG    0      0        0 eth2
10.0.0.0        0.0.0.0         255.255.255.0   U     0      0        0 eth1
10.1.0.9        10.0.0.1        255.255.255.255 UGH   0      0        0 eth1
169.254.0.0     0.0.0.0         255.255.0.0     U     0      0        0 eth0
================================================
ERROR: DNS not resolving cloudstack.apache.org
resolv.conf follows
nameserver 10.0.0.15
nameserver 10.1.0.9
nameserver 1.1.1.1
nameserver 8.8.8.8
root@s-2-VM:~#
```

**Reply by DaanHoogland** [reply]
ok, removing the route does nothing probably. your internal DNS server 10.0.0.15, can't even be reached from your ssvm. It should be reachable through eth2 by the looks of your data above. This is a nested env, so how is net 10.0.0.0 configured on your hypervisor? (and is it /24?)

**Reply by dylan62370** [reply]
Yes, it's a /24.
I created a cloudbr0 bridge on the physical port ens33

Here's the ip a of the server containing ssvm and cloudmanagement

```
[root@cloudstack-test ~]# ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: ens33: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq master cloudbr0 state UP group default qlen 1000
    link/ether 00:50:56:84:33:70 brd ff:ff:ff:ff:ff:ff
    altname enp2s1
3: cloudbr0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 00:50:56:84:33:70 brd ff:ff:ff:ff:ff:ff
    inet 10.0.0.22/24 brd 10.0.0.255 scope global noprefixroute cloudbr0
       valid_lft forever preferred_lft forever
    inet6 fe80::9897:46f5:48ea:f99c/64 scope link noprefixroute 
       valid_lft forever preferred_lft forever
4: cloud0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether fe:00:a9:fe:1a:b9 brd ff:ff:ff:ff:ff:ff
    inet 169.254.0.1/16 scope global cloud0
       valid_lft forever preferred_lft forever
    inet6 fe80::44ef:94ff:fed4:48a1/64 scope link 
       valid_lft forever preferred_lft forever
5: vnet0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master cloud0 state UNKNOWN group default qlen 1000
    link/ether fe:00:a9:fe:1a:b9 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::fc00:a9ff:fefe:1ab9/64 scope link 
       valid_lft forever preferred_lft forever
6: vnet1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master cloud0 state UNKNOWN group default qlen 1000
    link/ether fe:00:a9:fe:1c:3c brd ff:ff:ff:ff:ff:ff
    inet6 fe80::fc00:a9ff:fefe:1c3c/64 scope link 
       valid_lft forever preferred_lft forever
7: vnet2: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master cloudbr0 state UNKNOWN group default qlen 1000
    link/ether fe:00:11:00:00:07 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::fc00:11ff:fe00:7/64 scope link 
       valid_lft forever preferred_lft forever
8: vnet3: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master cloudbr0 state UNKNOWN group default qlen 1000
    link/ether fe:00:9e:00:00:05 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::fc00:9eff:fe00:5/64 scope link 
       valid_lft forever preferred_lft forever
9: vnet4: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master cloudbr0 state UNKNOWN group default qlen 1000
    link/ether fe:00:26:00:00:0c brd ff:ff:ff:ff:ff:ff
    inet6 fe80::fc00:26ff:fe00:c/64 scope link 
       valid_lft forever preferred_lft forever
10: vnet5: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master cloudbr0 state UNKNOWN group default qlen 1000
    link/ether fe:00:b2:00:00:0b brd ff:ff:ff:ff:ff:ff
    inet6 fe80::fc00:b2ff:fe00:b/64 scope link 
       valid_lft forever preferred_lft forever
[root@cloudstack-test ~]# 
```

**Reply by DaanHoogland** [reply]
and can 10.0.0.15 ping 10.0.0.66?

**Reply by dylan62370** [reply]
No :/

![image](https://github.com/user-attachments/assets/a6c6c134-70bc-42c5-8859-b8b58b37afe1)

**Comment by dylan62370** [comment]
If it helps, here's my ifconfig

```
[root@cloudstack-test ~]# ifconfig
cloud0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 169.254.0.1  netmask 255.255.0.0  broadcast 0.0.0.0
        inet6 fe80::44ef:94ff:fed4:48a1  prefixlen 64  scopeid 0x20<link>
        ether fe:00:a9:fe:1a:b9  txqueuelen 1000  (Ethernet)
        RX packets 50324  bytes 2267148 (2.1 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 109840  bytes 235208210 (224.3 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

cloudbr0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.0.0.22  netmask 255.255.255.0  broadcast 10.0.0.255
        inet6 fe80::9897:46f5:48ea:f99c  prefixlen 64  scopeid 0x20<link>
        ether 00:50:56:84:33:70  txqueuelen 1000  (Ethernet)
        RX packets 409893  bytes 139455270 (132.9 MiB)
        RX errors 0  dropped 34  overruns 0  frame 0
        TX packets 170736  bytes 39386362 (37.5 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

ens33: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        ether 00:50:56:84:33:70  txqueuelen 1000  (Ethernet)
        RX packets 246624  bytes 133031360 (126.8 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 351781  bytes 43201129 (41.1 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```

**Comment by dylan62370** [comment]
Hello @DaanHoogland 

I reinstalled everything today!
And I did well, because from my 2 SSVM VMs I can ping google.fr and other domains without a problem!

However, it's still impossible for me to download ISOs or Templates, even if I import them from my PC.

Here's the error message if I download them directly from cloudstack:

<img width="1233" alt="image" src="https://github.com/user-attachments/assets/018620f4-030d-4ac9-9448-3c0ef63e84b1">

**Comment by dylan62370** [comment]
IP 10.1.0.22 is the IP of my KVM hypervisor, which contains cloudstack-management, cloudstack-agent and NFS.

`2024-10-14 22:04:36,078 INFO  [storage.resource.NfsSecondaryStorageResource] (agentRequest-Handler-5:null) Determined host 10.1.0.22 corresponds to IP 10.1.0.22
2024-10-14 22:05:36,148 INFO  [storage.resource.NfsSecondaryStorageResource] (agentRequest-Handler-2:null) Determined host 10.1.0.22 corresponds to IP 10.1.0.22
2024-10-14 22:06:29,040 INFO  [storage.resource.NfsSecondaryStorageResource] (agentRequest-Handler-1:null) Determined host 10.1.0.22 corresponds to IP 10.1.0.22
2024-10-14 22:06:29,057 INFO  [storage.template.HttpTemplateDownloader] (agentRequest-Handler-1:null) No credentials configured for host=cdimage.debian.org:443
2024-10-14 22:06:29,243 ERROR [storage.template.HttpTemplateDownloader] (pool-1-thread-3:null) Failed to download https://cdimage.debian.org/images/cloud/bookworm/20241004-1890/debian-12-generic-amd64-20241004-1890.qcow2 due to redirection, response code: 302
2024-10-14 22:06:29,246 INFO  [storage.template.DownloadManagerImpl] (pool-1-thread-3:null) Download Completion for jobId: 2186d0a6-d96c-4dfd-99b9-942c97764789, status=UNRECOVERABLE_ERROR
2024-10-14 22:06:29,246 INFO  [storage.template.DownloadManagerImpl] (pool-1-thread-3:null) local: /mnt/SecStorage/511b42bf-1810-312b-9594-22ded64b3d01/template/tmpl/2/205/dnld3561603664234222540tmp_, bytes=(0 bytes) 0, error=Failed to download https://cdimage.debian.org/images/cloud/bookworm/20241004-1890/debian-12-generic-amd64-20241004-1890.qcow2 due to redirection, response code: 302, pct=0
2024-10-14 22:06:36,216 INFO  [storage.resource.NfsSecondaryStorageResource] (agentRequest-Handler-3:null) Determined host 10.1.0.22 corresponds to IP 10.1.0.22
2024-10-14 22:07:36,279 INFO  [storage.resource.NfsSecondaryStorageResource] (agentRequest-Handler-5:null) Determined host 10.1.0.22 corresponds to IP 10.1.0.22
2024-10-14 22:08:36,349 INFO  [storage.resource.NfsSecondaryStorageResource] (agentRequest-Handler-2:null) Determined host 10.1.0.22 corresponds to IP 10.1.0.22
2024-10-14 22:09:36,420 INFO  [storage.resource.NfsSecondaryStorageResource] (agentRequest-Handler-1:null) Determined host 10.1.0.22 corresponds to IP 10.1.0.22
2024-10-14 22:10:36,487 INFO  [storage.resource.NfsSecondaryStorageResource] (agentRequest-Handler-5:null) Determined host 10.1.0.22 corresponds to IP 10.1.0.22
2024-10-14 22:11:36,557 INFO  [storage.resource.NfsSecondaryStorageResource] (agentRequest-Handler-4:null) Determined host 10.1.0.22 corresponds to IP 10.1.0.22
2024-10-14 22:12:36,624 INFO  [storage.resource.NfsSecondaryStorageResource] (agentRequest-Handler-2:null) Determined host 10.1.0.22 corresponds to IP 10.1.0.22
2024-10-14 22:13:36,694 INFO  [storage.resource.NfsSecondaryStorageResource] (agentRequest-Handler-3:null) Determined host 10.1.0.22 corresponds to IP 10.1.0.22
2024-10-14 22:14:36,764 INFO  [storage.resource.NfsSecondaryStorageResource] (agentRequest-Handler-1:null) Determined host 10.1.0.22 corresponds to IP 10.1.0.22
2024-10-14 22:15:36,831 INFO  [storage.resource.NfsSecondaryStorageResource] (agentRequest-Handler-5:null) Determined host 10.1.0.22 corresponds to IP 10.1.0.22
2024-10-14 22:16:36,898 INFO  [storage.resource.NfsSecondaryStorageResource] (agentRequest-Handler-4:null) Determined host 10.1.0.22 corresponds to IP 10.1.0.22
2024-10-14 22:17:36,964 INFO  [storage.resource.NfsSecondaryStorageResource] (agentRequest-Handler-2:null) Determined host 10.1.0.22 corresponds to IP 10.1.0.22
2024-10-14 22:18:37,034 INFO  [storage.resource.NfsSecondaryStorageResource] (agentRequest-Handler-3:null) Determined host 10.1.0.22 corresponds to IP 10.1.0.22
2024-10-14 22:19:37,097 INFO  [storage.resource.NfsSecondaryStorageResource] (agentRequest-Handler-1:null) Determined host 10.1.0.22 corresponds to IP 10.1.0.22
2024-10-14 22:20:37,165 INFO  [storage.resource.NfsSecondaryStorageResource] (agentRequest-Handler-5:null) Determined host 10.1.0.22 corresponds to IP 10.1.0.22
2024-10-14 22:21:37,233 INFO  [storage.resource.NfsSecondaryStorageResource] (agentRequest-Handler-4:null) Determined host 10.1.0.22 corresponds to IP 10.1.0.22
2024-10-14 22:22:22,849 INFO  [storage.resource.NfsSecondaryStorageResource] (agentRequest-Handler-5:null) Determined host 10.1.0.22 corresponds to IP 10.1.0.22
2024-10-14 22:22:22,863 INFO  [storage.template.HttpTemplateDownloader] (agentRequest-Handler-5:null) No credentials configured for host=cdimage.debian.org:443
2024-10-14 22:22:23,040 ERROR [storage.template.HttpTemplateDownloader] (pool-1-thread-4:null) Failed to download https://cdimage.debian.org/debian-cd/current/amd64/iso-cd/debian-12.7.0-amd64-netinst.iso due to redirection, response code: 302
2024-10-14 22:22:23,042 INFO  [storage.template.DownloadManagerImpl] (pool-1-thread-4:null) Download Completion for jobId: 57f57729-347d-4721-b4ac-baa9a7a77ec5, status=UNRECOVERABLE_ERROR
2024-10-14 22:22:23,042 INFO  [storage.template.DownloadManagerImpl] (pool-1-thread-4:null) local: /mnt/SecStorage/511b42bf-1810-312b-9594-22ded64b3d01/template/tmpl/2/207/dnld1455001308766254764tmp_, bytes=(0 bytes) 0, error=Failed to download https://cdimage.debian.org/debian-cd/current/amd64/iso-cd/debian-12.7.0-amd64-netinst.iso due to redirection, response code: 302, pct=0
2024-10-14 22:22:37,299 INFO  [storage.resource.NfsSecondaryStorageResource] (agentRequest-Handler-4:null) Determined host 10.1.0.22 corresponds to IP 10.1.0.22
2024-10-14 22:23:37,361 INFO  [storage.resource.NfsSecondaryStorageResource] (agentRequest-Handler-5:null) Determined host 10.1.0.22 corresponds to IP 10.1.0.22
2024-10-14 22:24:37,473 INFO  [storage.resource.NfsSecondaryStorageResource] (agentRequest-Handler-2:null) Determined host 10.1.0.22 corresponds to IP 10.1.0.22
2024-10-14 22:25:37,537 INFO  [storage.resource.NfsSecondaryStorageResource] (agentRequest-Handler-3:null) Determined host 10.1.0.22 corresponds to IP 10.1.0.22
2024-10-14 22:26:37,601 INFO  [storage.resource.NfsSecondaryStorageResource] (agentRequest-Handler-1:null) Determined host 10.1.0.22 corresponds to IP 10.1.0.22`

<img width="1231" alt="image" src="https://github.com/user-attachments/assets/0bf6af12-6d90-4a55-9239-b665e950efd0">

**Reply by rajujith** [reply]
@dylan62370  can you check whether the default centos template is downloaded, also try registering another URL.

**Comment by dylan62370** [comment]
Hello,

The CentOS template is Ready
![image](https://github.com/user-attachments/assets/72f64de5-df21-417f-8539-3eb4ed26b0df)

I just tested with CentOS 7 and Ubuntu 24.04 and got no errors.
I only have this error for Debian though, I can download the ISO from my PC or from my cloudstack server over SSH without any problems.

**Comment by dylan62370** [comment]
Since VM SSVM1 no worries either @rajujith 
![image](https://github.com/user-attachments/assets/66d64662-a2d3-423a-b3bb-5d4272b00b73)

I've just noticed another error
![image](https://github.com/user-attachments/assets/2f61c384-52cf-4219-8413-437d46d299fc)

**Reply by weizhouapache** [reply]
@dylan62370 
can you use http URL instead of https ?

**Comment by DaanHoogland** [comment]
closed this for lack of interaction, please reopen or open a new item if applicable
